gateway-js/src/index.ts

```diff
@@ -451,11 +415,6 @@ export class ApolloGateway implements GraphQLService {
         healthCheck: this.externalSubgraphHealthCheckCallback.bind(this),
         getDataSource: this.externalGetDataSourceCallback.bind(this),
       });
-      if (!result?.supergraphSdl) {
```

Comment:
**note** This is somewhat of a silly check, since much worse schemas could be passed, and are caught later (e.g. `" "` won't fail here but will fail when the schema is parsed later).

So I decided to move this check later in the stack, and make it a GraphQL error https://github.com/apollographql/federation/pull/1812/files#diff-1a556fe2fd6a545a37d601c8185534fec07e931e2599b32d71c767cd149df20eR447, since it's an invalid schema (oddly, our parse code doesn't handle that case and throws an obscure error...I should file an issue about that).

Reply:
Does `trim()`ming in the meantime help with that or does parse throw funky on empty string too?

Reply:
Well, after some digging, the cause of the error was we ended up passing `null` as the supergraph schema if we got a falsy supergraph schema during load (i.e. we'd make a `""` into a `null`). So that was the cause of the weird parsing errors.

I've fixed those up and added tests for the types of errors in https://github.com/apollographql/federation/pull/1812/commits/c62b4a04c8c977f31a7154bbe4421e5854e1ca34